functions/function/http-request-function/src/main/java/org/springframework/cloud/fn/http/request/HttpRequestFunctionProperties.java

```diff
@@ -35,64 +34,64 @@
  */
 @Validated
 @ConfigurationProperties("http.request")
-public class HttpRequestFunctionProperties implements HttpRequestProperties {
-	private static final HttpMethod DEFAULT_HTTP_METHOD = HttpMethod.GET;
+public class HttpRequestFunctionProperties {
 
 	private static final Class<?> DEFAULT_RESPONSE_TYPE = String.class;
+
 	/**
-	 * The URL to issue an http request to, as a static value.
+	 * The type used to interpret the response.
 	 */
-	private String url;
+	private Class<?> expectedResponseType = DEFAULT_RESPONSE_TYPE;
 
 	/**
-	 * The (static) request body; if neither this nor bodyExpression is provided, the payload
-	 * will be used.
+	 * Request timeout in milliseconds.
 	 */
-	private Object body;
+	private long timeout = 30_000;
 
 	/**
-	 * The kind of http method to use.
+	 * A SpEL expression against incoming message to determine the URL to use.
 	 */
-	private HttpMethod httpMethod = DEFAULT_HTTP_METHOD;
+	private Expression urlExpression;
 
 	/**
-	 * The type used to interpret the response.
+	 * A SpEL expression to derive the request method from the incoming message.
 	 */
-	private Class<?> expectedResponseType = DEFAULT_RESPONSE_TYPE;
+	private Expression httpMethodExpression = new SpelExpressionParser().parseExpression("'GET'");
 
 	/**
-	 * Request timeout in milliseconds.
+	 * A SpEL expression to derive the request body from the incoming message.
 	 */
-	private long timeout = 30_000;
+	private Expression bodyExpression;
 
 	/**
-	 * A Map of HTTP request headers.
+	 * A SpEL expression used to derive the http headers map to use.
 	 */
-	private HttpHeaders headers = new HttpHeaders();
+	private Expression headersExpression;
 
-	@NotEmpty
-	@Override
-	public String getUrl() {
-		return this.url;
-	}
+	/**
+	 * A SpEL expression used to compute the final result, applied against the whole http
+	 * {@link org.springframework.http.ResponseEntity}.
+	 */
+	private Expression replyExpression = new SpelExpressionParser().parseExpression("body");
```

Comment:
Can we use a plain `new FunctionExpression<HttpEntity<?>>(HttpEntity::getBody)` for optimization?